src/Raven.Server/Documents/ETL/EtlLoader.cs

```diff
@@ -492,5 +491,60 @@ public void HandleDatabaseValueChanged(DatabaseRecord record)
                 }
             }
         }
+
+        public string TombstoneCleanerIdentifier => $"ETL loader for {_database.Name}";
+
+        public Dictionary<string, long> GetLastProcessedTombstonesPerCollection()
+        {
+            var lastProcessedTombstones = new Dictionary<string, long>(StringComparer.OrdinalIgnoreCase);
+
+            var ravenEtls = _databaseRecord.RavenEtls;
+            var sqlEtls = _databaseRecord.SqlEtls;
+
+            foreach (var config in ravenEtls) 
+                MarkTombstonesForDeletion(config, lastProcessedTombstones);
+
+            foreach (var config in sqlEtls)
+                MarkTombstonesForDeletion(config, lastProcessedTombstones);
+
+            return lastProcessedTombstones;
+        }
+
+        private void MarkTombstonesForDeletion<T>(EtlConfiguration<T> config, Dictionary<string, long> lastProcessedTombstones) where T : ConnectionString
+        {
+            foreach (var transform in config.Transforms)
+            {
+                var state = EtlProcess.GetProcessState(_database, config.Name, transform.Name);
+                var etag = ChangeVectorUtils.GetEtagById(state.ChangeVector, _database.DbBase64Id);
```

Comment:
Why you no longer use `EtlProcessState.GetLastProcessedEtagForNode` ?

Reply:
This dictionary is updated on the responsible node.
How the other nodes should know if it is okay to delete their tombstones?

Reply:
Makes sense.